simplefs/simplefs.go

```diff
@@ -219,10 +220,28 @@ func (k *SimpleFS) doCopy(ctx context.Context, srcPath, destPath keybase1.Path)
 	defer dst.Close()
 
 	if src.Type() == keybase1.DirentType_FILE || src.Type() == keybase1.DirentType_EXEC {
-		_, err = io.Copy(dst, src)
+		err = copyWithCancellation(ctx, dst, src)
+		if err != nil {
+			return err
+		}
 	}
 
-	return err
+	return nil
+}
+
+func copyWithCancellation(ctx context.Context, dst io.Writer, src io.Reader) error {
+	for {
+		if err := ctx.Err(); err != nil {
+			return err
+		}
+		_, err := io.CopyN(dst, src, 64*1024)
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
```

Comment:
This uses CopyN for which ```On return, written == n if and only if err == nil.```.

However on reflection using CopyBuffer is better for the given types.